src/components/include/utils/threads/thread.h

```diff
@@ -172,7 +173,11 @@ class Thread {
    */
   void stop();
 
-  void join();
+  /**
+   * @brief Starts join procedure.
+   * @param[in] force_stop If true stop() is called first otherwise is skipped.
+   */
+  void join(const bool force_stop = true);
```

Comment:
@orhan-mehmedov please don't use default values. It is really unreadable in the code.
Also I would rather suggest to create some `enum` instead of boolean variable